app/src/main/java/com/doublesp/coherence/adapters/ListCompositionArrayAdapter.java

```diff
@@ -161,22 +160,23 @@ private void saveToFireBase() {
                 ideaList);
     }
 
-    private void saveItemsToFireBase(int start, int count) {
+    private void updateItemInFireBase(int start, int count) {
         Plan plan = mIdeaInteractor.getPlan();
         for (int i = 0; i < count; i++) {
             int pos = start + count - 1;
             Idea updatedIdea = mIdeaInteractor.getIdeaAtPos(pos);
-//            mShoppingListDatabaseReference.child(mIdeaInteractor.getPlan().getId()).child(
-//                    ConstantsAndUtils.IDEAS).child(String.valueOf(pos)).setValue(updatedIdea);
+            mShoppingListDatabaseReference.child(plan.getId()).child(ConstantsAndUtils.IDEAS)
+                    .child(String.valueOf(pos)).setValue(updatedIdea);
         }
     }
 
     private void saveNewItemsToFireBase(int start, int count) {
         Plan plan = mIdeaInteractor.getPlan();
         for (int i = 0; i < count; i++) {
-            int pos = start + count - 1;
+            int pos = start + count - 2;
```

Comment:
This is what i wanted you to check, i verified it works, but good for you to confirm count is one more than expected.